### PR TITLE
Add SignalCore driver

### PR DESCRIPTION
**Pull request by edumur**
All in the title I guess ^^ .

### CODE REVIEW

**Comment by edumur** [comment]
That's interesting, mypy passed on my laptop:
```bash
python -m mypy --ignore-missing-import --no-warn-unused-ignores -p qcodes_contrib_drivers
Success: no issues found in 156 source files
```

**Comment by astafan8** [comment]
> That's interesting, mypy passed on my laptop:

likely because you have Windows? and the CI is using linux. mypy understands platform differences so we can fix it, let me find out how ....

File: qcodes_contrib_drivers/drivers/SignalCore/SignalCore.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        name (str): Name of the instrument.
+        dll_path (str): Path towards the instrument DLL.
+        """
+        
+        (super().__init__)(name, **kwargs)
+        
+        # Adapt the path to the computer language
+        dll_path = os.path.join(os.environ['PROGRAMFILES'], dll_path)
+
+        self._dll = ctypes.WinDLL(dll_path)
```
Comment (astafan8) [inline qcodes_contrib_drivers/drivers/SignalCore/SignalCore.py:134]: so let's use this https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks here - and if the platform is not windows then raise a nice error to the user (mypy will also like it and understand it)

File: docs/examples/SignalCore.ipynb [hunk 2]
```diff
@@ -0,0 +1,207 @@
+{
```
Comment (astafan8) [inline docs/examples/SignalCore.ipynb:1]: there seems to be an issue with the notebook as well that it's not a valid JSON, see here https://github.com/QCoDeS/Qcodes_contrib_drivers/blob/d8b66cd63692e22267ec29a2910516f8ce5cb18d/docs/examples/SignalCore.ipynb

Comment (edumur) [reply docs/examples/SignalCore.ipynb:1]: Yep that's me adding:
```json

            "nbsphinx":{
               "execute":"never"
            }
```
too fast...
Thanks for the catch!

File: qcodes_contrib_drivers/drivers/SignalCore/SignalCore.py (hunk 1 above)
Comment (edumur) [reply qcodes_contrib_drivers/drivers/SignalCore/SignalCore.py:134]: Nice, I am implementing it, thanks!

**Comment by edumur** [comment]
Well, I tried my best (can't make local test first)

**Comment by edumur** [comment]
ok, I am out of ideas...
Does someone has a hint?

File: qcodes_contrib_drivers/drivers/SignalCore/SignalCore.py (hunk 1 above)
Comment (astafan8) [reply qcodes_contrib_drivers/drivers/SignalCore/SignalCore.py]: should be done now

**Review by astafan8 — APPROVED** [review]

**Comment by edumur** [comment]
Well, great thank for all that work!

**Pull request merged**